code/app/src/main/java/com/project/lordofthewings/Models/Wallet/Wallet.java

```diff
@@ -41,9 +51,9 @@ public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                     if (document.exists()) {
                         Log.d(TAG, "DocumentSnapshot data: " + document.getData());
                         username = Objects.requireNonNull(document.get("username")).toString();
-                        score = Integer.parseInt(Objects.requireNonNull(document.get("Score")).toString());
-                        qrCodesCount = Integer.parseInt(Objects.requireNonNull(document.get("numberOfQRCodes")).toString());
-                        qrCodes = (ArrayList<String>) document.get("QRCodes");
+                        //score = Integer.parseInt(Objects.requireNonNull(document.get("Score")).toString());
+                        //qrCodesCount = Integer.parseInt(Objects.requireNonNull(document.get("numberOfQRCodes")).toString());
+                        qrCodes = (ArrayList<QRCode>) document.get("QRCodes");
```

Comment:
This line will give an error since in firebase the QRCodes in the user document are an array of Strings, whereas here it is expected to return an array of Object QRCodes. We can store an object in an array in firebase.

Reply:
"the array is a supported data type in Firestore. It’s very useful for storing data in String format or even custom objects, as long as the size of the document is less than 1 MiB (1,048,576 bytes)." I read this online, we should be able to have an arrayList filled with QRCode objects.
https://medium.com/firebase-tips-tricks/how-to-update-an-array-of-objects-in-firestore-cdb611a56073#:~:text=Even%20from%20the%20beginning%2C%20the,more%20likely%20to%20be%20used.

Reply:
Sure I will try testing it then on the scanQRCode page.

Reply:
try it and lmk, if it does not work I'll have to do this
https://stackoverflow.com/questions/71759027/retrieving-arraylist-of-custom-objects-in-firebase